spec/ready_set_spec.rb

```diff
@@ -259,5 +259,24 @@
           Readyset.configuration.shard
       end
     end
+
+    # NOTE: If query tags aren't available, it will annotate anything.
+    # Adding a feature toggle via config would be redundant.
```

Comment:
I think we'll want to package up this logging config neatly so users of the gem can enable it without much hassle. What are your thoughts on doing something like this in a railtie?
```ruby
initializer 'readyset.query_annotator' do |app|
  ActiveSupport.on_load(:after_intialization) do
    if Readyset.config.enable_query_annotations?
        if Rails.configuration.active_record.query_log_tags_enabled
          Rails.configuration.active_record.query_log_tags ||= []
          Rails.configuration.active_record.query_log_tags << {
            routed_to_readyset?: ->(context) do
              Readyset::QueryAnnotator.routing_to_readyset?
            end,
          }
      else
        Rails.configuration.active_record.query_log_tags_enabled = true
        Rails.configuration.active_record.query_log_tags = [{
          routed_to_readyset?: ->(context) do
              Readyset::QueryAnnotator.routing_to_readyset?
            end,
          }]
      end
    end
  end
end
```
Basically, if query log tagging is already enabled, we just add our special attribute to the list of tags, and if it's not already enabled, we set the list of tags to consist of _only_ our tag. What do you think?

I want to make the gem as easy to use as possible with as little configuration as possible 😄

Reply:
I was on the fence about this due to changing things on their Rails app. But appending the Readyset context if they have `query_log_tags_enabled` is an excellent idea. 

I'll implement this.

Reply:
I was considering this, but decided against it to avoid changing their application state "invisibly". However, I think your example actually makes it work out.

I think it's a good idea to append our context to `query_log_tags` after checking if they have `query_log_tags_enabled`. However, I don't think we should enable it for them, at most, we should warn them that it is disabled/unavailable.

This is a great quality-of-life suggestion.

Reply:
Currently having difficulties with Combustion due to how the initializer order works. It's difficult to test the negative condition of when query tags aren't available. May refactor the Combustion initialization into a support file or something.

Thoughts welcome.

Reply:
For now, it's just logging warn as a placeholder. What it should really do is use the `Logger` that I gotta make already lol.

Reply:
@helpotters Yeah, I hear you about not changing the application's config. Maybe it would be more palatable if we were to only enable the query log tags when `Rails.env == 'environment'`?

I'm just worried that people won't enable this feature unless they read the docs, and I think it has the potential to make the gem a lot easier to use.